### PR TITLE
[receiver/fluentforward] Release port on shutdown

### PR DESCRIPTION
**Pull request by gagansingh355**
This is fix for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/9110.

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

**Review by jpkrohling — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Comment by gagansingh355** [comment]
@jpkrohling , I have resolved the merge conflict. Please merge this.

**Comment by gagansingh355** [comment]
Seems like some changes were pushed in CHANGELOG.md after i resolved previous conflicts. Please merge this again.

File: CHANGELOG.md [hunk 1]
```diff
@@ -68,6 +68,7 @@
 - `signalfxexporter`: Fix bug to enable timeouts for correlating traces and metrics (#9101)
 - `windowsperfcountersreceiver`: fix exported values being integers instead of doubles (#9138)
 - `prometheusreceiver`: Fix issues with relabelling the `job` and `instance` labels. (#8780)
+- `fluentforwardreceiver`: Fluentforward receiver not releasing port for reloading of OTEL collector (#9111)
```
Comment (jpkrohling) [inline CHANGELOG.md:71]: Sorry for one more change about this, but we just had a release and this has to be moved to the unreleased section. I tried updating this PR, but I couldn't push my changes.

Comment (gagansingh355) [reply CHANGELOG.md]: I have update this push to unreleased section.

**Comment by gagansingh355** [comment]
@jpkrohling , @dmitryax New code was pushed in main hence test were failing, I have down merged main branch into this. Please merge this.

**Comment by codeboten** [comment]
Thanks @gagansingh355

**Pull request merged**